perception/lidar_transfusion/lib/preprocess/voxel_generator.cpp

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lidar_transfusion/preprocess/voxel_generator.hpp"
+
+#include <sensor_msgs/point_cloud2_iterator.hpp>
+
+namespace lidar_transfusion
+{
+
+VoxelGenerator::VoxelGenerator(
+  const DensificationParam & densification_param, const TransfusionConfig & config,
+  cudaStream_t & stream)
+: config_(config), stream_(stream)
+{
+  pd_ptr_ = std::make_unique<PointCloudDensification>(densification_param);
+  pre_ptr_ = std::make_shared<PreprocessCuda>(config_, stream_);
+  cloud_data_d_ = cuda::make_unique<unsigned char[]>(config_.cloud_capacity_ * MAX_CLOUD_STEP_SIZE);
+  points_cpu_.resize(config_.cloud_capacity_ * config_.num_point_feature_size_);
+  affine_past2current_d_ = cuda::make_unique<float[]>(AFF_MAT_SIZE);
+  pre_ptr_.reset(new PreprocessCuda(config_, stream_));
```

Comment:
Please avoid using new and use unique ptr when possible

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/6890/commits/48200d207d459c01f9fb7dc38b33b355bc1437f0